nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/CountText.java

```diff
@@ -0,0 +1,318 @@
+        return relationships;
+    }
+
+    @OnScheduled
+    public void onSchedule(ProcessContext context) {
+        this.countLines = context.getProperty(TEXT_LINE_COUNT_PD).isSet()
+                ? context.getProperty(TEXT_LINE_COUNT_PD).asBoolean() : false;
+        this.countLinesNonEmpty = context.getProperty(TEXT_LINE_NONEMPTY_COUNT_PD).isSet()
+                ? context.getProperty(TEXT_LINE_NONEMPTY_COUNT_PD).asBoolean() : false;
+        this.countWords = context.getProperty(TEXT_WORD_COUNT_PD).isSet()
+                ? context.getProperty(TEXT_WORD_COUNT_PD).asBoolean() : false;
+        this.countCharacters = context.getProperty(TEXT_CHARACTER_COUNT_PD).isSet()
+                ? context.getProperty(TEXT_CHARACTER_COUNT_PD).asBoolean() : false;
+        this.splitWordsOnSymbols = context.getProperty(SPLIT_WORDS_ON_SYMBOLS_PD).isSet()
+                ? context.getProperty(SPLIT_WORDS_ON_SYMBOLS_PD).asBoolean() : false;
+        this.characterEncoding = context.getProperty(CHARACTER_ENCODING_PD).getValue();
+    }
+
+    /**
+     * Will count text attributes of the incoming stream.
+     */
+    @Override
+    public void onTrigger(ProcessContext context, ProcessSession processSession) throws ProcessException {
+        FlowFile sourceFlowFile = processSession.get();
+        if (sourceFlowFile == null) {
+            return;
+        }
+        AtomicBoolean error = new AtomicBoolean();
+
+        lineCount = 0;
+        lineNonEmptyCount = 0;
+        wordCount = 0;
+        characterCount = 0;
+
+        processSession.read(sourceFlowFile, in -> {
+            long start = System.nanoTime();
+
+            // Iterate over the lines in the text input
+            try {
+                BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(in, characterEncoding));
+                String line;
+                while ((line = bufferedReader.readLine()) != null) {
+                    if (countLines) {
+                        lineCount++;
+                    }
+
+                    if (countLinesNonEmpty) {
+                        if (line.trim().length() > 0) {
+                            lineNonEmptyCount++;
+                        }
+                    }
+
+                    if (countWords) {
+                        wordCount += countWordsInLine(line, splitWordsOnSymbols);
+                    }
+
+                    if (countCharacters) {
+                        characterCount += line.length();
+                    }
+                }
+                long stop = System.nanoTime();
+                if (getLogger().isDebugEnabled()) {
+                    final long durationNanos = stop - start;
+                    DecimalFormat df = new DecimalFormat("#.###");
+                    getLogger().debug("Computed metrics in " + durationNanos + " nanoseconds (" + df.format(durationNanos / 1_000_000_000.0) + " seconds).");
+                }
+                String message = generateMetricsMessage();
```

Comment:
Given that the default log level for processors is WARN, perhaps we should avoid all of this concatenation if log level of info is not enabled?

Reply:
Will do.